pkg/services/publicdashboards/service/service_test.go

```diff
@@ -1295,36 +1295,64 @@ func assertOldValueIfNull(t *testing.T, expectedValue bool, oldValue bool, nulla
 }
 
 func TestDeletePublicDashboard(t *testing.T) {
+	pubdash := &PublicDashboard{Uid: "2", OrgId: 1, DashboardUid: "uid"}
+
+	type mockResponse struct {
+		PublicDashboard *PublicDashboard
+		Err             error
+	}
 	testCases := []struct {
 		Name             string
 		AffectedRowsResp int64
 		ExpectedErrResp  error
 		StoreRespErr     error
+		mockStore        *mockResponse
 	}{
 		{
-			Name:             "Successfully deletes a public dashboards",
+			Name:             "Successfully deletes a public dashboard",
 			AffectedRowsResp: 1,
 			ExpectedErrResp:  nil,
 			StoreRespErr:     nil,
+			mockStore:        &mockResponse{pubdash, nil},
 		},
 		{
-			Name:             "Public dashboard not found",
+			Name:             "Public dashboard not found by UID",
 			AffectedRowsResp: 0,
-			ExpectedErrResp:  nil,
+			ExpectedErrResp:  ErrInternalServerError.Errorf("Delete: failed to find public dashboard by uid: pubdashUID: error"),
 			StoreRespErr:     nil,
+			mockStore:        &mockResponse{pubdash, errors.New("error")},
 		},
 		{
-			Name:             "Database error",
+			Name:             "Public dashboard not found by database",
 			AffectedRowsResp: 0,
-			ExpectedErrResp:  ErrInternalServerError.Errorf("Delete: failed to delete a public dashboard by Uid: uid db error!"),
+			ExpectedErrResp:  ErrPublicDashboardNotFound.Errorf("Delete: public dashboard not found by uid: pubdashUID"),
+			StoreRespErr:     nil,
+			mockStore:        &mockResponse{nil, nil},
+		},
+		{
+			Name:             "Public dashboard UID does not belong to the dashboard",
+			AffectedRowsResp: 0,
+			ExpectedErrResp:  ErrInvalidUid.Errorf("Delete: the public dashboard does not belong to the dashboard"),
+			StoreRespErr:     nil,
+			mockStore:        &mockResponse{&PublicDashboard{Uid: "2", OrgId: 1, DashboardUid: "wrong"}, nil},
+		},
+
+		{
+			Name:             "Failed to delete - Database error",
+			AffectedRowsResp: 1,
+			ExpectedErrResp:  ErrInternalServerError.Errorf("Delete: failed to delete a public dashboard by Uid: pubdashUID db error!"),
 			StoreRespErr:     errors.New("db error!"),
+			mockStore:        &mockResponse{pubdash, nil},
 		},
 	}
 
 	for _, tt := range testCases {
 		t.Run(tt.Name, func(t *testing.T) {
 			store := NewFakePublicDashboardStore(t)
-			store.On("Delete", mock.Anything, mock.Anything).Return(tt.AffectedRowsResp, tt.StoreRespErr)
+			store.On("Find", mock.Anything, mock.Anything).Return(tt.mockStore.PublicDashboard, tt.mockStore.Err)
+			if tt.ExpectedErrResp == nil || tt.StoreRespErr != nil {
```

Comment:
I added this conditon as goroutine throws this error 
```The code you are testing needs to make 1 more call(s).``` as as ```store.on("Delete")``` is never called if the validations before it fail